hubmap_api_py_client/internal.py

```diff
@@ -44,19 +45,19 @@ def _fill_request_dict(
             input_type: str, output_type: str, input_set: List[str],
             genomic_modality: str, p_value: float):
         request_dict = {'input_type': input_type, 'input_set': input_set}
-        if input_type == 'gene' and output_type == 'cell':
+        if input_type == 'gene' and output_type in ['cell', 'cluster', 'organ'] or input_type in \
+                ['cluster', 'organ'] and output_type == 'gene':
             request_dict['genomic_modality'] = genomic_modality
-        if input_type in ['organ', 'gene'] and output_type == 'gene':
+            request_dict['logical_operator'] = "and"
+        if (input_type in ['organ', 'cluster'] and output_type == 'gene'
+                or input_type == 'gene' and output_type in ['organ', 'cluster']):
```

Comment:
I think this pre-flight validation is too much already, and I prefer not to expand it. My preference would be to get rid of all of it, and if there's an error from the API, raise an exception then.

Reply:
Works for me.  Will drop this.